Source/Orts.Simulation/Simulation/RollingStocks/TrainCar.cs

```diff
@@ -819,6 +821,81 @@ public virtual void Initialize()
+
+                if (WagonType != WagonTypes.Engine)   // if car is not a locomotive then determine wheelbase
+                {
+                    if (Bogies < 2)  // if less then two bogies assume that it is a fixed wheelbase wagon
+                    {
+                        if (Axles == 2)
+                        {
+                            RigidWheelBaseM = 3.5052f;       // Assume a standard 4 wheel (2 axle) wagon - wheel base - 11' 6" (3.5052m)
+                        }
+                        else if (Axles == 3)
+                        {
+                            RigidWheelBaseM = 3.6576f;       // Assume a standard 6 wheel (3 axle) wagon - wheel base - 12' 2" (3.6576m)
+                        }
+                    }
+                    else if (Bogies == 2)
+                    {
+                        if (BogieSize == 2)
+                        {
+                            if (WagonType == WagonTypes.Passenger)
+                            {
+                                RigidWheelBaseM = 2.4384f;       // Assume a standard 4 wheel passenger bogie (2 axle) wagon - wheel base - 8' (2.4384m)
+                            }
+                            else
+                            {
+                                RigidWheelBaseM = 1.6764f;       // Assume a standard 4 wheel freight bogie (2 axle) wagon - wheel base - 5' 6" (1.6764m)
+                            }
+                        }
+                        else if (BogieSize == 3)
+                        {
+                            RigidWheelBaseM = 3.6576f;       // Assume a standard 6 wheel bogie (3 axle) wagon - wheel base - 12' 2" (3.6576m)
+                        }
+                    }
+                }
+                if (WagonType == WagonTypes.Engine)   // if car is a locomotive and either a diesel or electric then determine wheelbase
+                {
+                    if (EngineType != EngineTypes.Steam)  // Assume that it is a diesel or electric locomotive
+                    {
+                        if (BogieSize == 2)
+                        {
+                            RigidWheelBaseM = 1.6764f;       // Set a default in case no option is found - assume a standard 4 wheel (2 axle) bogie - wheel base - 5' 6" (1.6764m)
+                        }
+                        else if (BogieSize == 3)
+                        {
+                            RigidWheelBaseM = 3.5052f;       // Assume a standard 6 wheel bogie (3 axle) locomotive - wheel base - 11' 6" (3.5052m)
+                        }
+                    }
+                    else // assume steam locomotive
+                    {
+
+                        if (LocoNumDrvAxles >= Axles) // Test to see if ENG file value is too big (typically doubled)
+                        {
+                            LocoNumDrvAxles = LocoNumDrvAxles / 2;  // Appears this might be the number of wheels rather then the axles.
```

Comment:
Agree with the aim here, but would be helpful to write a message to the logfile.

Reply:
Message added.